src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/filter/FilterClause.java

```diff
@@ -12,6 +19,111 @@
     example = """
              {"name": "Aaron", "country": "US"}
               """)
-public record FilterClause(LogicalExpression logicalExpression) {}
+public record FilterClause(LogicalExpression logicalExpression) {
+  public void validate(CommandContext commandContext) {
+    DocumentProjector indexingProjector = commandContext.indexingProjector();
+    // If nothing specified, everything indexed
+    if (indexingProjector.isIdentityProjection()) {
+      return;
+    }
+    validateLogicalExpression(logicalExpression, indexingProjector);
+  }
 
-// public record FilterClause(List<ComparisonExpression> comparisonExpressions) {}
+  public void validateLogicalExpression(
+      LogicalExpression logicalExpression, DocumentProjector indexingProjector) {
+    for (LogicalExpression subLogicalExpression : logicalExpression.logicalExpressions) {
+      validateLogicalExpression(subLogicalExpression, indexingProjector);
+    }
+    for (ComparisonExpression subComparisonExpression : logicalExpression.comparisonExpressions) {
+      validateComparisonExpression(subComparisonExpression, indexingProjector);
+    }
+  }
+
+  public void validateComparisonExpression(
+      ComparisonExpression comparisonExpression, DocumentProjector indexingProjector) {
+    String path = comparisonExpression.getPath();
+    boolean isPathIndexed = indexingProjector.isPathIncluded(path);
+
+    // If path is "_id" and it's denied, the operator can only be $eq or $in
+    if (path.equals(DocumentConstants.Fields.DOC_ID) && !isPathIndexed) {
+      FilterOperator filterOperator = comparisonExpression.getFilterOperations().get(0).operator();
+      // if operator is not $eq or $in, throw error
+      if (!filterOperator.equals(ValueComparisonOperator.EQ)
```

Comment:
Can simplify comparisons since `ValueComparisonOperator` is enum so no need for `equals()`.
So maybe:

```
if (filterOperator == ValueComparisonOperator.EQ || filterOperator == ValueComparisonOperator.IN) {
  return;
}
// otherwise throw JsonApiException
```